presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -286,11 +289,103 @@ public void testEnum()
         }
     }
 
-    @Test
-    public void testTimestamp()
+    @Test(dataProvider = "testTimestampDataProvider")
+    public void testTimestamp(boolean legacyTimestamp, boolean insertWithPresto)
     {
-        // TODO timestamp is not correctly read (see comment in StandardColumnMappings.timestampColumnMappingUsingSqlTimestamp)
-        // testing this is hard because of https://github.com/prestodb/presto/issues/7122
+        LocalDateTime beforeEpoch = LocalDateTime.of(1958, 1, 1, 13, 18, 3, 123_000_000);
+        LocalDateTime epoch = LocalDateTime.of(1970, 1, 1, 0, 0, 0);
+        LocalDateTime afterEpoch = LocalDateTime.of(2019, 03, 18, 10, 01, 17, 987_000_000);
+
+        ZoneId jvmZone = ZoneId.systemDefault();
+
+        LocalDateTime timeGapInJvmZone = LocalDateTime.of(1970, 1, 1, 0, 13, 42);
```

Comment:
I'd name this `timeGapInJvmZone1` so it's more obvious there are two and they look symmetric